tests/test_workspace.py

```diff
@@ -910,3 +910,10 @@ def test_workspace_without_validation(mocker, simplemodels_model_data):
 
     pyhf.Workspace(dict(ws), validate=False)
     assert pyhf.schema.validate.called is False
+
+
+def test_workspace_invalid_specification():
+    spec = {"channels": [{"name": "SR", "samples_wrong_name": []}]}
+    # ensure that an invalid specifications gets caught as such
```

Comment:
```suggestion
    # Ensure that an invalid specifications gets caught as such
    # before a KeyError: 'samples' could be reached.
```

Reply:
This is all great, but I'd just add a comment to make it clear in the future what this test adds.